commitserver/commit/commit.go

```diff
@@ -118,10 +120,15 @@ func (s *Service) handleCommitRequest(logCtx *log.Entry, r *apiclient.CommitHydr
 		return out, "", fmt.Errorf("failed to checkout target branch: %w", err)
 	}
 
-	logCtx.Debug("Clearing repo contents")
-	out, err = gitClient.RemoveContents()
-	if err != nil {
-		return out, "", fmt.Errorf("failed to clear repo: %w", err)
+	logCtx.Debug("Clearing repo path contents")
+	for _, path := range r.Paths {
+		if hydrator.IsRootPath(path.Path) {
+			return "", "", fmt.Errorf("refusing to clear repo root path: %q", path.Path)
+		}
```

Comment:
Let's move this validation up to the top. We should entirely reject requests where any path targets the repo root.

Reply:
PTAL now.